### PR TITLE
Remove imperative API's

### PR DESCRIPTION
**Pull request by DylanPiercey**
## Description
Removes all imperative API's and exposing properties/events on the DOM as described in #732.

## References
Implements #732 

This again is semver major, but I'll rebase as necessary.

### CODE REVIEW

**Comment by DylanPiercey** [comment]
@agliga yeah I was waiting to rebase until all the patches went through. I am working on it now.

**Comment by agliga** [comment]
> @agliga yeah I was waiting to rebase until all the patches went through. I am working on it now.

👍

File: src/components/ebay-infotip/index.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
-const emitAndFire = require('../../common/emit-and-fire');
 
 module.exports = require('marko-widgets').defineComponent({
```
Comment (seangates) [inline src/components/ebay-infotip/index.js]: Minor nit: there's whitespace at the beginning of this file.

**Comment by DylanPiercey** [comment]
@seangates I looked through but I’ll take another look. I’m pretty sure all of this was undocumented besides marking properties as stateful in the attribute table.

File: src/components/ebay-infotip/index.js (hunk 1 above)
Comment (DylanPiercey) [reply src/components/ebay-infotip/index.js]: Our linter should catch this 😞

**Pull request merged**